test/API/trefer.c

```diff
@@ -2018,96 +2018,98 @@ test_reference_region_1D(H5F_libver_t libver_low, H5F_libver_t libver_high)
 static void
 test_reference_obj_deleted(void)
 {
-#ifndef NO_REFERENCE_TO_DELETED
+    //#ifndef NO_REFERENCE_TO_DELETED
     hid_t fid1;          /* HDF5 File IDs            */
     hid_t dataset,       /* Dataset ID               */
         dset2;           /* Dereferenced dataset ID  */
     hid_t      sid1;     /* Dataspace ID             */
     H5R_ref_t  oref;     /* Object Reference to test */
     H5O_type_t obj_type; /* Object type              */
     herr_t     ret;      /* Generic return value     */
-#endif
+                         //#endif
     MESSAGE(5, ("Testing References to Deleted Objects - SKIPPED for now due to no support\n"));
-#ifndef NO_REFERENCE_TO_DELETED
-    /* Create file */
-    fid1 = H5Fcreate(FILE_REF_OBJ_DEL, H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
-    CHECK(fid1, H5I_INVALID_HID, "H5Fcreate");
+    //#ifndef NO_REFERENCE_TO_DELETED
+    if ((vol_cap_flags_g & H5VL_CAP_FLAG_REF_BASIC) && (vol_cap_flags_g & H5VL_CAP_FLAG_FILE_BASIC)) {
```

Comment:
For this particular test I think we should check for H5VL_CAP_FLAG_DATASET_BASIC and H5VL_CAP_FLAG_LINK_BASIC as well.

Reply:
I can add those checks in.